modules/openxBidAdapter.js

```diff
@@ -1,298 +1,330 @@
+  let hasCustomParam = false;
+  bids.forEach(function (bid) {
+    if (bid.params.customParams) {
+      let customParamsForBid = utils._map(Object.keys(bid.params.customParams), customKey => formatCustomParms(customKey, bid.params.customParams));
+      let formattedCustomParams = window.btoa(customParamsForBid.join('&'));
+      hasCustomParam = true;
+      customParamsForAllBids.push(formattedCustomParams);
+    } else {
+      customParamsForAllBids.push('');
     }
+  });
+  if (hasCustomParam) {
+    oxParams.tps = customParamsForAllBids.join(',');
   }
 
-  function callBids(params) {
-    let isIfr;
-    const bids = params.bids || [];
-    let currentURL = (window.parent !== window) ? document.referrer : window.location.href;
-    currentURL = currentURL && encodeURIComponent(currentURL);
-    try {
-      isIfr = window.self !== window.top;
-    } catch (e) {
-      isIfr = false;
-    }
-    if (bids.length === 0) {
-      return;
+  let customFloorsForAllBids = [];
+  let hasCustomFloor = false;
+  bids.forEach(function (bid) {
+    if (bid.params.customFloor) {
+      customFloorsForAllBids.push(bid.params.customFloor * 1000);
+      hasCustomFloor = true;
+    } else {
+      customFloorsForAllBids.push(0);
     }
+  });
+  if (hasCustomFloor) {
+    oxParams.aumfs = customFloorsForAllBids.join(',');
+  }
 
-    let delDomain = bids[0].params.delDomain;
-    let bcOverride = bids[0].params.bc;
+  let url = `//${delDomain}/w/1.0/arj`;
+  return {
+    method: 'GET',
+    url: url,
+    data: oxParams,
+    payload: {'bids': bids, 'startTime': new Date()}
+  };
+}
+
+function buildOXVideoRequest(bids) {
+  return bids.map(function(bid) {
+    let url = 'http://' + bid.params.delDomain + '/v/1.0/avjp';
+    let oxVideoParams = generateVideoParameters(bid);
+    return {
+      method: 'GET',
+      url: url,
+      data: oxVideoParams,
+      payload: {'bid': bid, 'startTime': new Date()}
+    };
+  });
+}
+
+function generateVideoParameters(bid) {
+  let oxVideo = bid.params.video;
+  let oxVideoParams = {
+    auid: bid.params.unit,
+    url: oxVideo.url
+  };
 
-    startTime = new Date(params.start);
-    if (params.timeout) {
-      timeout = params.timeout;
-    }
-    if (bids[0].params.hasOwnProperty('sendBoPixel') && typeof (bids[0].params.sendBoPixel) === 'boolean') {
-      shouldSendBoPixel = bids[0].params.sendBoPixel;
+  if (oxVideo.vtest) {
+    oxVideoParams.vtest = oxVideo.vtest;
+  }
+  if (oxVideo.be) {
+    oxVideoParams.be = oxVideo.be;
+  }
+
+  return oxVideoParams;
+}
+
+function createVideoBidResponses(response, {bid, startTime}) {
+  let bidResponses = [];
+
+  if (response !== undefined && response.cache_key !== '') {
+    let bidResponse = {};
+    bidResponse.requestId = bid.bidId;
+    bidResponse.bidderCode = BIDDER_CODE;
+    // default 5 mins
+    bidResponse.ttl = 300;
+    // true is net, false is gross
+    bidResponse.netRevenue = true;
+    // waiting for this to be exposed
```

Comment:
Maybe remove the comment here would be better, I did on my end

Reply:
sure